rotkehlchen/history/events/structures/evm_swap.py

```diff
@@ -0,0 +1,141 @@
+                HistoryEventSubType.RECEIVE,
+                HistoryEventSubType.FEE,
+            ],
+            asset: Asset,
+            amount: 'FVal',
+            location_label: str | None = None,
+            notes: str | None = None,
+            identifier: int | None = None,
+            counterparty: str | None = None,
+            product: EvmProduct | None = None,
+            address: 'ChecksumEvmAddress | None' = None,
+            extra_data: dict[str, Any] | None = None,
+            event_identifier: str | None = None,
+    ):
+        """Combines EvmEvent with SwapEvent to represent evm swaps."""
+        super().__init__(
+            tx_hash=tx_hash,
+            sequence_index=sequence_index,
+            timestamp=timestamp,
+            location=location,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=event_subtype,
+            asset=asset,
+            amount=amount,
+            location_label=location_label,
+            notes=notes,
+            identifier=identifier,
+            counterparty=counterparty,
+            product=product,
+            address=address,
+            extra_data=extra_data,
+            event_identifier=event_identifier,
+        )
+
+    @property
+    def entry_type(self) -> HistoryBaseEntryType:
+        return HistoryBaseEntryType.EVM_SWAP_EVENT
+
+    @classmethod
+    def deserialize_from_db(cls: type['EvmSwapEvent'], entry: tuple) -> 'EvmSwapEvent':
+        """Deserialize a EvmSwapEvent DB tuple.
+        May raise:
+        - DeserializationError
+        - UnknownAsset
+        But these exceptions shouldn't normally happen since
+        the data from the db should already be correct.
+        """
+        entry = cast('EVM_EVENT_DB_TUPLE_READ', entry)
+        amount = deserialize_fval(entry[7], 'amount', 'evm swap event')
+        return cls(
+            identifier=entry[0],
+            event_identifier=entry[1],
+            sequence_index=entry[2],
+            timestamp=TimestampMS(entry[3]),
+            location=Location.deserialize_from_db(entry[4]),
+            location_label=entry[5],
+            asset=Asset(entry[6]).check_existence(),
+            amount=amount,
+            notes=entry[8] or None,
+            event_subtype=HistoryEventSubType.deserialize(entry[10]),  # type: ignore  # should always be correct from the DB
+            extra_data=cls.deserialize_extra_data(entry=entry, extra_data=entry[11]),
+            tx_hash=deserialize_evm_tx_hash(entry[12]),
+            counterparty=entry[13],
+            product=EvmProduct.deserialize(entry[14]) if entry[14] is not None else None,
+            address=deserialize_optional(input_val=entry[15], fn=string_to_evm_address),
+        )
+
+    def serialize(self) -> dict[str, Any]:
+        """Serialize the event for api."""
+        serialized_data = super().serialize()  # Super is EvmEvent.
+        serialized_data['auto_notes'] = self._generate_auto_notes()
```

Comment:
I believe this is possible and more explicit

```suggestion
        serialized_data = super(EvmEvent, self).serialize()
        serialized_data['auto_notes'] = self._generate_auto_notes()
```

Reply:
actually `super(EvmEvent, self)` gets the next one back after EvmEvent (SwapEvent), which is apparently the expected behavior of python's method resolution order. 

But I've changed this to `EvmEvent.serialize(self)` instead since that also makes it more explicit.